app/controllers/users_controller.rb

```diff
@@ -13,16 +21,34 @@ def new
   end
 
   def create
-    user = User.new(user_params)
-    
-    authenticate!(:password) if user.save
-    after_login_redirect
+    @user = User.new(user_params)
+    redirect_url = nil
+
+    if @user.save
+      authenticate!(:password)
+      @user.send_confirmation
+    else
+      redirect_to(signup_url, notice: "There was a problem") && return
+    end
+    redirect_url = accept_invite_url(params[:token]) if params[:token].present?
+    after_login_redirect(redirect_url)
```

Comment:
move lines 33 and 34 in the if statement and then in the else write render :new to show the form again with the filled details and errors for each input.